uptime.go

```diff
@@ -0,0 +1,324 @@
+// See: https://docs.digitalocean.com/reference/api/api-reference/#tag/Projects
+type UptimeChecksService interface {
+	List(context.Context, *ListOptions) ([]UptimeCheck, *Response, error)
+	Get(context.Context, string) (*UptimeCheck, *Response, error)
+	GetState(context.Context, string) (*UptimeCheck, *Response, error)
+	Create(context.Context, *CreateUptimeCheckRequest) (*UptimeCheck, *Response, error)
+	Update(context.Context, string, *UpdateUptimeCheckRequest) (*UptimeCheck, *Response, error)
+	Delete(context.Context, string) (*Response, error)
+	GetAlert(context.Context, string, string) (*Alert, *Response, error)
+	ListAlerts(context.Context, *ListOptions, string) ([]Alert, *Response, error)
+	CreateAlert(context.Context, *CreateAlertRequest, string) (*Alert, *Response, error)
+	UpdateAlert(context.Context, string, string, *UpdateAlertRequest) (*Alert, *Response, error)
+	DeleteAlert(context.Context, string, string) (*Response, error)
+}
+
+// UptimeChecksServiceOp handles communication with Uptime Check methods of the DigitalOcean API.
+type UptimeChecksServiceOp struct {
+	client *Client
+}
+
+// UptimeCheck represents a DigitalOcean UptimeCheck configuration.
+type UptimeCheck struct {
+	ID      string   `json:"id"`
+	Name    string   `json:"name"`
+	Type    string   `json:"type"`
+	Target  string   `json:"target"`
+	Regions []string `json:"regions"`
+	Enabled bool     `json:"enabled"`
+}
+
+// Alert represents a DigitalOcean Alert configuration.
+type Alert struct {
+	ID            string         `json:"id"`
+	Name          string         `json:"name"`
+	Type          string         `json:"type"`
+	Threshold     int            `json:"threshold"`
+	Comparison    string         `json:"comparison"`
+	Notifications *Notifications `json:"notifications"`
+	Period        string         `json:"period"`
+}
+
+// CreateUptimeAlertRequest represents the request to create a new alert.
+type CreateAlertRequest struct {
+	Name          string         `json:"name"`
+	Type          string         `json:"type"`
+	Threshold     int            `json:"threshold"`
+	Comparison    string         `json:"comparison"`
+	Notifications *Notifications `json:"notifications"`
+	Period        string         `json:"period"`
+}
+
+// UpdateUptimeAlertRequest represents the request to create a new alert.
+type UpdateAlertRequest struct {
+	Name          string         `json:"name"`
+	Type          string         `json:"type"`
+	Threshold     int            `json:"threshold"`
+	Comparison    string         `json:"comparison"`
+	Notifications *Notifications `json:"notifications"`
+	Period        string         `json:"period"`
+}
+
+// Notifications represents a DigitalOcean Notifications configuration.
+type Notifications struct {
+	Email []string `json:"email"`
+	Slack []Slack  `json:"slack"`
+}
+
+// Slack represents a DigitalOcean Slack configuration.
+type Slack struct {
+	Channel string `json:"channel"`
+	URL     string `json:"url"`
+}
```

Comment:
Maybe we can just share the `SlackDetails` type from monitoring?

https://github.com/digitalocean/godo/blob/16f4ba922214ad74129b1a4136d88dc94edaa402/monitoring.go#L100-L104